aptos-move/framework/aptos-framework/sources/genesis.move

```diff
@@ -184,15 +188,97 @@ module aptos_framework::genesis {
+        let num_employee_groups = vector::length(&employees);
+        let unique_accounts = vector::empty();
+
+        while (i < num_employee_groups) {
+            let j = 0;
+            let employee_group = vector::borrow(&employees, i);
+            let num_employees_in_group = vector::length(&employee_group.accounts);
+
+            let buy_ins = simple_map::create();
+
+            while (j < num_employees_in_group) {
+                let account = vector::borrow(&employee_group.accounts, j);
+                assert!(
+                    !vector::contains(&unique_accounts, account),
+                    error::already_exists(EDUPLICATE_ACCOUNT),
+                );
+                vector::push_back(&mut unique_accounts, *account);
+
+                let employee = create_signer(*account);
+                let total = coin::balance<AptosCoin>(*account);
+                let coins = coin::withdraw<AptosCoin>(&employee, total);
+                simple_map::add(&mut buy_ins, *account, coins);
+
+                j = j + 1;
+            };
+
+            let j = 0;
+            let num_vesting_events = vector::length(&employee_group.vesting_schedule_numerator);
+            let schedule = vector::empty();
+
+            while (j <  num_vesting_events) {
+                let numerator = vector::borrow(&employee_group.vesting_schedule_numerator, j);
+                let event = fixed_point32::create_from_rational(*numerator, employee_group.vesting_schedule_denominator);
+                vector::push_back(&mut schedule, event);
+
+                j = j + 1;
+            };
+
+            let vesting_schedule = vesting::create_vesting_schedule(
+                schedule,
+                1663456089, // Update before mainnet or pass in by config
+                30 * 24 * 60 * 60, // 30 days
+            );
+
+            let admin = employee_group.validator.validator_config.owner_address;
+            vesting::create_vesting_contract(
+                &create_signer(admin),
+                &employee_group.accounts,
+                buy_ins,
+                vesting_schedule,
+                admin,
+                employee_group.validator.validator_config.operator_address,
+                employee_group.validator.validator_config.voter_address,
+                employee_group.validator.commission_percentage,
+                x"",
+            );
+
+            i = i + 1;
+        }
     }
 
     fun create_initialize_validators_with_commission(
-        _aptos_framework: &signer,
-        _validators: vector<ValidatorConfigurationWithCommission>) {
+        aptos_framework: &signer,
+        validators: vector<ValidatorConfigurationWithCommission>,
+    ) {
+        let i = 0;
+        let num_validators = vector::length(&validators);
+        let unique_accounts = vector::empty();
+
+        while (i < num_validators) {
+std::debug::print(&b"a");
+            let validator = vector::borrow(&validators, i);
+
+            assert!(
```

Comment:
This would fail if we have a single staker account with multiple staking contracts with different operators. That's actually a valid use case.

Reply:
I suspect you can fixup in your next PR? We can also add more complexity to my test case at the bottom. Since most of this code is reused, I feel fairly certain that we can just spit out the mainnet genesis transaction to a json blob and validate it. We'll have to do that anyway. Plus we'll have a few other operational tests.

Reply:
Okay